sled-agent/src/sled_agent.rs

```diff
@@ -309,11 +336,138 @@ impl SledAgent {
         )
         .await?;
 
-        Ok(SledAgent { id, storage, instances, lazy_nexus_client, services })
+        let sled_agent = SledAgent {
+            inner: Arc::new(SledAgentInner {
+                id,
+                addr: sled_address,
+                storage,
+                instances,
+                hardware,
+                services,
+                lazy_nexus_client,
+
+                // TODO(https://github.com/oxidecomputer/omicron/issues/1917):
+                // Propagate usage of this request queue throughout the Sled Agent.
+                //
+                // Also, we could maybe de-dup some of the backoff code in the request queue?
+                nexus_request_queue: NexusRequestQueue::new(),
+            }),
+        };
+
+        // We immediately add a notification to the request queue about our
+        // existence. If inspection of the hardware later informs us that we're
+        // actually running on a scrimlet, that's fine, the updated value will
+        // be received by Nexus eventually.
+        sled_agent.notify_nexus_about_self(&log);
+
+        // Begin monitoring the underlying hardware, and reacting to changes.
+        let sa = sled_agent.clone();
+        tokio::spawn(async move {
+            sa.monitor(log).await;
+        });
+
+        Ok(sled_agent)
+    }
+
+    async fn monitor(&self, log: Logger) {
+        // Start monitoring the hardware for changes
+        let mut hardware_updates = self.inner.hardware.monitor();
+
+        // Scan the existing system for noteworthy events
+        // that may have happened before we started monitoring
+        if self.inner.hardware.is_scrimlet() {
```

Comment:
Finding an attached tofino driver tells us that we're running on a scrimlet, but it doesn't mean we have all the resources needed to launch a switch zone.  Specifically, we need the `tfpkt0` link to exist.  Without that, `tfportd` can't create `tfport`s, so `mgs` and maghemite can't run.  In practice, I would expect that link to exist long before `sled-agent` runs, but it's at least a theoretical concern.

Reply:
I fixed this in https://github.com/oxidecomputer/omicron/pull/1933 , when actually launching the switch zone